marketing/tasks.py

```diff
@@ -92,7 +98,38 @@ def send_followup_emails():
 
     # Try sending all the followup emails
     for args in mails_to_send:
-        render_and_dispatch_email(*args)
+        template, subscriber = args
+        render_and_dispatch_email(template, subscriber)
+        sent_emails[template.name].add(subscriber.user.email)
+
+    send_report(sent_emails)
```

Comment:
@arjunsinghy96, I understand your reasoning to call this task right after the `send_followup_emails` task. But I don't think they are actually dependent on each other. I have two scenarios in my mind where this can be problematic.

1. Say in the future we want to run the `send_followup_emails` task multiple times in a day. In that case, this approach will not work.
2. If `send_followup_emails` fails (say a timeout or something else) after sending a couple of emails, we will be losing those in our report. Right?

Therefore, I think it would be better to keep them separate. What do you think?
cc @giovannicimolin

Reply:
@shimulch
> Say in the future we want to run the `send_followup_emails` task multiple times in a day. In that case, this approach will not work.

I think we should be sending the report after every trigger of `send_followup_emails` as that would provide a better report of what was sent.
If in future we decide to run `send_followup_emails` multiple time, we can either
1. Send report after every run (recommended)
2. Add a separate task to fetch emails from `SentEmail` model and trigger `send_report`

> If send_followup_emails fails (say a timeout or something else) after sending a couple of emails, we will be losing those in our report. Right?

Correct. Thanks for catching this. I think putting it in a `try..finally` block will fix. I will update the PR with it.